### PR TITLE
refactor(e2e-tests): import the instantsearch-e2e-tests repo

### PR DESCRIPTION
**Pull request by Haroenv**
based on the state after https://github.com/algolia/instantsearch-e2e-tests/pull/38 and its review

NOTE: this PR _must_ be **rebased** (not squashed), keeping the original history to have a working `git blame`

FX-1964

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -115,6 +115,50 @@ module.exports = {
...
+      ],
+      extends: ['plugin:wdio/recommended'],
+      plugins: ['wdio'],
+      env: {
+        jasmine: true,
+      },
+      rules: {
+        // we are exporting test "factories", so we need to allow export
+        'jest/no-export': 0,
+        // we use jasmine
```
Comment (sarahdayan) [inline .eslintrc.js:151]: Is it required to use Jasmine with WebdriverIO? Could we do another PR to use Jest instead and get rid of it?

File: tests/e2e/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
+## [2.0.2](https://github.com/algolia/instantsearch-e2e-tests/compare/v2.0.1...v2.0.2) (2022-02-03)
```
Comment (sarahdayan) [inline tests/e2e/CHANGELOG.md:1]: Do we still need a changelog? This is now a private package, this doesn't seem useful any longer.

File: tests/e2e/LICENSE [hunk 3]
```diff
@@ -0,0 +1,21 @@
+The MIT License (MIT)
```
Comment (sarahdayan) [inline tests/e2e/LICENSE:1]: Do we still need a license?

File: tests/e2e/README.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
+# End-2-End test suite for InstantSearch
+
+This repository contains the end-2-end test suite for InstantSearch.
+
+## Installation
+
+The workspace automatically links the dependencies when you run `yarn` at the root of the repository.
```
Comment (sarahdayan) [inline tests/e2e/README.md:7]:
```suggestion
The workspace automatically links the dependencies when you run `yarn` at the root of the monorepo.
```

File: tests/e2e/README.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+## Running the test suite
+
+In addition to the test suite this package contains two [WebdriverIO](https://webdriver.io) configurations:
+
+- `local`: Run the test suite on [Chrome browser](https://google.com/chrome) on your local machine.
+- `saucelabs`: Run the test suite on multiple browsers on the [Sauce Labs service](https://saucelabs.com).
+
+All configurations will run a static web server on your machine on port `5000` to serve the content of the `website` directory in your project, containing the InstantSearch demos to run our tests against.
```
Comment (sarahdayan) [inline tests/e2e/README.md:16]: This is still worded as if this was a reusable package to install in your repo.

File: tests/e2e/README.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+    # ...
+    steps:
+      # ...
+      - store_test_results:
+        path: junit/wdio
+```
+
+# License
+
+instantsearch-e2e-tests is licensed under the [MIT License](http://www.opensource.org/licenses/mit-license.php).
```
Comment (sarahdayan) [inline tests/e2e/README.md:93]: This is obsolete no?

File: tests/e2e/CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,97 @@
+# Contributing
```
Comment (sarahdayan) [inline tests/e2e/CONTRIBUTING.md:1]: Do we still need a contributing guide?

File: tests/e2e/flavors.js [hunk 8]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+  flavors: ['js'],
+};
```
Comment (sarahdayan) [inline tests/e2e/flavors.js:3]: Any way we can use an ES module or won't that work with WebdriverIO configs?

File: tests/e2e/package.json [hunk 9]
```diff
@@ -0,0 +1,32 @@
+{
+  "name": "@instantsearch/e2e-tests",
+  "private": true,
+  "version": "3.0.0",
+  "description": "end-to-end test suite for InstantSearch",
```
Comment (sarahdayan) [inline tests/e2e/package.json:5]:
```suggestion
  "description": "End-to-end test suite for InstantSearch",
```

File: .eslintrc.js (hunk 1 above)
Comment (Haroenv) [reply .eslintrc.js:151]: Jasmine is the closest to jest's expect, but you can't use jest

File: tests/e2e/CHANGELOG.md (hunk 2 above)
Comment (Haroenv) [reply tests/e2e/CHANGELOG.md:1]: It's actually more work to prevent the changelog than to have it (you need to ignore it in gitignore), so I kept it. It's indeed arguable whether it's useful since we have git diff too.

File: tests/e2e/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (Haroenv) [reply tests/e2e/README.md:16]: website is the built directory, examples is the source

File: tests/e2e/README.md (hunk 6 above)
Comment (Haroenv) [reply tests/e2e/README.md:93]: No, this part is still relevant and used in this project

File: tests/e2e/flavors.js (hunk 8 above)
Comment (Haroenv) [reply tests/e2e/flavors.js:3]: the config needs to be cjs to work without transpiling. Probably it could be esm if we set up the whole monorepo as esm, but I don't want to do that until everything is migrated, as there's always some tooling that doesn't fully work with it yet (can think of jest for example)

File: tests/e2e/CONTRIBUTING.md (hunk 7 above)
Comment (Haroenv) [reply tests/e2e/CONTRIBUTING.md:1]: yes, it explains how to write tests

File: tests/e2e/README.md (hunk 5 above)
Comment (Haroenv) [reply tests/e2e/README.md:16]: It's still how we use the tests in this repo :)

File: .eslintrc.js (hunk 1 above)
Comment (sarahdayan) [reply .eslintrc.js:151]: It looks like it's possible but there's no straightforward path: https://github.com/webdriverio/webdriverio/issues/2052

Anyway, let's consider it some other time.

File: tests/e2e/CHANGELOG.md (hunk 2 above)
Comment (Haroenv) [reply tests/e2e/CHANGELOG.md:8]: It actually will be written on every release, just like the other packages :)

File: tests/e2e/LICENSE (hunk 3 above)
Comment (Haroenv) [reply tests/e2e/LICENSE:1]: I don't know the legal intricacies, but I don't think you can remove a license?

File: tests/e2e/CONTRIBUTING.md (hunk 7 above)
Comment (sarahdayan) [reply tests/e2e/CONTRIBUTING.md:1]: That should be in our [main contributing guide](https://github.com/algolia/instantsearch.js/blob/master/CONTRIBUTING.md#tests) then.

File: tests/e2e/README.md (hunk 6 above)
Comment (Haroenv) [reply tests/e2e/README.md:93]: It's now linked from the main contributing as well :)

File: tests/e2e/LICENSE (hunk 3 above)
Comment (sarahdayan) [reply tests/e2e/LICENSE:1]: Technically this new package is pure addition in this repo, and it's not even exposed as a public package.

Also, the whole monorepo is MIT so this directory inherits it.

File: tests/e2e/CHANGELOG.md (hunk 2 above)
Comment (sarahdayan) [reply tests/e2e/CHANGELOG.md:8]: Ah, I didn't realize we'd be doing one changelog per project.

Unrelated, but do you know how this will look like in [releases](https://github.com/algolia/instantsearch.js/releases)?

Comment (Haroenv) [reply tests/e2e/CHANGELOG.md:8]: I hadn't gone through the whole release flow, it's possible only one of the packages will have a GitHub release, but normally it should be one release per tag, and that tag having its changelog. Every package gets its own tag

File: tests/e2e/README.md (hunk 5 above)
Comment (sarahdayan) [reply tests/e2e/README.md:16]: Not exactly because you no longer install it per repo. It's installed in the monorepo and will work with every new flavor that we migrate, without changes to any configuration (aside from adding new flavors to `flavors.js`) no?

I guess what I'm saying is that there's nothing more to install or set up for end-to-end tests to work in this repo. I'm not sure whether this information is still useful.

File: tests/e2e/CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+      'LG - 4.5 Cu. Ft. 8-Cycle High-Efficiency Top-Loading Washer - White',
+      'Samsung - 4.2 Cu. Ft. 9-Cycle High-Efficiency Steam Front-Loading Washer - Platinum',
+      'Panasonic - 1.3 Cu. Ft. Mid-Size Microwave - Stainless steel/black/silver',
+      'LG - 2.0 Cu. Ft. Mid-Size Microwave - Black Stainless',
+    ]);
+  });
+});
+```
+
+General guidelines when writing tests:
```
Comment (sarahdayan) [inline tests/e2e/CONTRIBUTING.md:82]:
```suggestion
Here are some general guidelines when writing end-to-end tests:
```

File: tests/e2e/CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+
+- Each step should be translated as a `it` function, while this is not mandatory it helps to make the scenario more readable and properly [separate each actions in Sauce Labs reports](https://user-images.githubusercontent.com/13209/62311104-56217d80-b48b-11e9-94dc-3c18b9ddc2af.png).
```
Comment (sarahdayan) [inline tests/e2e/CONTRIBUTING.md:84]:
```suggestion
- Each separate step should be in its own `it` function. This isn't a hard rule, but it helps make the scenario more readable and properly [separate each action in Sauce Labs reports](https://user-images.githubusercontent.com/13209/62311104-56217d80-b48b-11e9-94dc-3c18b9ddc2af.png).
```

File: tests/e2e/CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+- All actions on the browser are asynchronous, so be sure to always `await` them. **Never run multiple asynchronous commands in parallel as it can confuse some browsers (Internet Explorer)**.
+- Use helper functions when possible, for readability but also for maintainability (if one widget is updated, we only have to updates its helpers without touching the tests). [More about Helpers](#helpers).
```
Comment (sarahdayan) [inline tests/e2e/CONTRIBUTING.md:86]:
```suggestion
- Use [helper](#helpers) functions whenever possible, for readability but also for maintainability—if one widget is updated, we only have to updates its helpers without touching the tests.
```

File: tests/e2e/CONTRIBUTING.md [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+- Do not make assertions to know if an action was correctly performed in the browser. If an action fails (trying to click on an non-existing element for example) then WebdriverIO will automatically throw and fail the test, so asserting on it ourselves is redundant.
+- Only assert what you want to see on the page after an action (Is this checkbox selected? Is the result list correct? etc.)
```
Comment (sarahdayan) [inline tests/e2e/CONTRIBUTING.md:88]:
```suggestion
- Only assert what you want to see on the page after an action—for example, is the checkbox selected, is the result list correct, etc.
```

File: tests/e2e/CONTRIBUTING.md [hunk 14]
```diff
@@ -0,0 +1,97 @@
...
+- You may need to add some additional steps compared to the original scenario, to wait for some elements to update for example (this was not done in the example above for simplicity but you can find some examples [here](https://github.com/algolia/instantsearch-e2e-tests/blob/5a2456b8d63afa684931b0447f00df821b02752b/specs/brand-and-query.spec.ts#L14-L16) or [here](https://github.com/algolia/instantsearch-e2e-tests/blob/5a2456b8d63afa684931b0447f00df821b02752b/specs/price-range.spec.ts#L13-L22)).
```
Comment (sarahdayan) [inline tests/e2e/CONTRIBUTING.md:89]: The links are outdated.

**Review by sarahdayan — APPROVED** [review]
Some final nits, but this LGTM.
